### PR TITLE
feat: prevent focus on disabled dates

### PR DESCRIPTION
**Pull request by remischwartz**
Hi,

This is a proposal to prevent focus on disabled dates (cf #25 ).
 I struggled to find a satisfying behavior, and eventually settled on the following one :
- up/down/left/right arrows will look for the first available (not disabled) date in the corresponding direction, up to a maximum difference in days (customizable via the `max-search-days` attribute)
- Home/End will look for the first/last available date in the current month
- Page up/down will look for the first available date in the previous/next month
- Shift + Page up/down will look for the next available date a year before/after the current date

Please let me know what you think.

### CODE REVIEW

File: src/components/wc-datepicker/wc-datepicker.tsx [hunk 1]
```diff
@@ -647,7 +779,9 @@ export class WCDatepicker {
                             onMouseLeave={this.onMouseLeave}
                             role="gridcell"
                             tabIndex={
-                              isSameDay(day, this.currentDate) && !this.disabled
+                              isSameDay(day, this.currentDate) &&
+                              !this.disableDate(day) &&
```
Comment (Sqrrl) [inline src/components/wc-datepicker/wc-datepicker.tsx:783]: Suggestion: Use `isDisabled` from line 747 to not execute the `disableDate` function twice.

File: src/components/wc-datepicker/wc-datepicker.tsx [hunk 2]
```diff
@@ -350,38 +452,68 @@ export class WCDatepicker {
...
+        this.updateCurrentDate(
+          this.getAvailableDate(this.currentDate, 'nextYear'),
+          true
+        );
       } else {
-        this.updateCurrentDate(getNextMonth(this.currentDate), true);
+        this.updateCurrentDate(
+          this.getAvailableDate(this.currentDate, 'nextMonth'),
+          true
+        );
```
Comment (Sqrrl) [inline src/components/wc-datepicker/wc-datepicker.tsx:503]: Issue: Something seems off here. I've tried these settings: 
```js
datepicker.maxSearchDays = 1000;

datepicker.disableDate = (date) => {
  return (
    date.getDay() === 6 ||
    date.getDay() === 0 ||
    date.getDate() === 15 ||
    date.getYear() === 124
  );
};
```
- Select a day in December 2023
- Hit Page Down or Shift + Page Down
- ⛔️ Nothing happens

I assume the same Page Up has the same issue. Maybe one or two tests would be helpful.

**Review by Sqrrl — CHANGES_REQUESTED** [review]
Thanks for the PR. Found one or two issues.

File: src/components/wc-datepicker/wc-datepicker.tsx (hunk 2 above)
Comment (remischwartz) [reply src/components/wc-datepicker/wc-datepicker.tsx:503]: This has to do with the new behavior where disabled dates are no longer focusable.
When hitting Page Down or Shift + Page Down, the date picker will look for the next available date in next month or next year. But if there is none (in your exemple, the whole 2024 year is disabled), the currentDate is not changed.

Do you see another approach that could be taken ?

File: src/components/wc-datepicker/wc-datepicker.tsx (hunk 1 above)
Comment (remischwartz) [reply src/components/wc-datepicker/wc-datepicker.tsx]: I made the change ✅

**Review by Sqrrl — APPROVED** [review]

**Comment by Sqrrl** [comment]
Thanks again. Released as v0.6.0. https://github.com/Sqrrl/wc-datepicker/releases/tag/v0.6.0

**Pull request merged**